WordPress/gutenberg
# Add table block column alignment

## Description
Adds column alignment options to the table block. Closes #15823

**Dev decisions**
Code-wise, the part where I've had to break from the norm is by using a `data-align` attribute to store the alignment value for a cell. This is because the cell attributes are deeply nested and sourced using a `query` matcher, so normal attributes can't be used.

Applying an alignment adds a class name and `data-align` attribute to each table cell in a column. There aren't many alternatives to this approach that I can think of, apart from using inline styles. `colgroup`/`col` elements don't support text align, so they're out of the question.

## How has this been tested?
Added unit and e2e tests.

Manual testing steps:
1. Add a table block to a post
2. Add some text to the first column of the table.
3. Ensure one of the cells in the first column is selected
4. From the toolbar try each of the column alignment options
5. Verify that text in the each cell in the column is aligned in the editor
6. Preview the post
7. Verify that the text is aligned when viewing the post.

## Screenshots 
**Editor**
<img width="425" alt="Screen Shot 2019-07-30 at 10 46 50 am" src="https://user-images.githubusercontent.com/677833/62096810-7a2c6580-b2b7-11e9-9ed7-daa666eb7f99.png">

**Post**
<img width="466" alt="Screen Shot 2019-07-30 at 10 47 45 am" src="https://user-images.githubusercontent.com/677833/62096824-903a2600-b2b7-11e9-8b15-dac05e77d81b.png">

## Types of changes


New feature (non-breaking change which adds functionality)


## Checklist:
